desktop/cmp/treemap/TreeMap.js

```diff
@@ -0,0 +1,297 @@
+            chart: {margin: false},
+            credits: false,
+            title: false,
+            legend: {enabled: false},
+            exporting
+        };
+    }
+
+    getThemeConfig() {
+        return XH.darkTheme ? clone(DarkTheme) : clone(LightTheme);
+    }
+
+    getModelConfig() {
+        const {data, highchartsConfig, algorithm, tooltip} = this.model,
+            {defaultTooltip} = this;
+
+        return {
+            ...highchartsConfig,
+            tooltip: {
+                enabled: !!tooltip,
+                useHTML: true,
+                padding: 0,
+                shape: 'square',
+                shadow: false,
+                pointFormatter: function() {
+                    if (!tooltip) return;
+                    const {record} = this;
+                    return tooltip == true ? defaultTooltip(record) : tooltip(record);
+                }
+            },
+            series: [{
+                data,
+                type: 'treemap',
+                animation: false,
+                layoutAlgorithm: algorithm,
+                borderWidth: 0,
+                dataLabels: {
+                    enabled: true,
+                    allowOverlap: false,
+                    align: 'left',
+                    verticalAlign: 'top',
+                    style: {textOutline: 'none', visibility: 'hidden'}
+                },
+                events: {click: this.onClick}
+            }]
+        };
+    }
+
+    //----------------------
+    // Click handling
+    //----------------------
+    onClick = (e) => {
+        this._clickCount++;
+        this._debouncedClickHandler(e.point.record, e);
+        if (this._clickCount >= 2) this._debouncedClickHandler.flush();
+    };
+
+    clickHandler(record, e) {
+        const {onClick, onDoubleClick} = this.model;
+        if (onClick && this._clickCount === 1) {
+            onClick(record, e);
+        } else if (onDoubleClick && this._clickCount === 2) {
+            onDoubleClick(record, e);
+        }
+        this._clickCount = 0;
+    }
+
+    //----------------------
+    // Selection handling
+    //----------------------
+    syncSelection() {
+        if (!this._chart) return;
+
+        const {selectedIds, maxDepth, gridModel} = this.model;
+        let toSelect = [...selectedIds];
+
+        // Fallback to parent node if selection exceeds max depth
+        if (maxDepth && gridModel && gridModel.treeMode) {
+            toSelect = selectedIds.map(id => {
+                const delim = Cube.RECORD_ID_DELIMITER,
+                    parts = id.split(delim),
+                    idDepth = parts.filter(p => p !== 'root').length,
+                    toRemove = idDepth - maxDepth;
+
+                return toRemove < 1 ? id : parts.slice(0, -toRemove).join(delim);
+            });
+        }
+
+        // Update selection in chart
+        this._chart.series[0].data.forEach(node => {
+            node.select(toSelect.includes(node.id), true);
```

Comment:
I always wonder if it would be a worthwhile optimization to get the selected list into a Set and check via `has()` over `includes()`.  It might be. Again, being more paranoid than usual with these nested collection operations.